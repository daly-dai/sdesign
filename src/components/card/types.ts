import { CardProps } from 'antd';
import { ReactNode } from 'react';

export interface SCardProps extends Omit<CardProps, 'children'> {
  children?: ReactNode;
  hasBottomPadding?: boolean; // 底部是否包含边距
}
