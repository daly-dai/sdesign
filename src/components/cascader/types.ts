import { CascaderProps } from 'antd';

export type SingleValueType = (string | number)[];
export type ValueType = SingleValueType | SingleValueType[] | undefined;

export type SCascaderProps = Omit<
  CascaderProps<any>,
  'onChange' | 'value' | 'defaultValue' | 'multiple'
> & {
  onChange?: (value: string, selectOptions?: any[]) => void;
  value?: string | null | ValueType;
  options?: any[];
  defaultValue?: string | ValueType;
  multiple?: boolean;
};
