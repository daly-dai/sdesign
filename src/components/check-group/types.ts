import { Checkbox } from 'antd';
import { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import { ExtraComType } from '@dalydb/sdesign/types/base';

export type CheckboxGroupType = HTMLAttributes<object> &
  Omit<ComponentProps<typeof Checkbox.Group>, 'onChange' | 'value'>;
export type CheckboxValueType = any;

export type SCheckGroupProps = CheckboxGroupType &
  ExtraComType & {
    value?: CheckboxValueType[] | undefined | string;
    onChange?: (value: string | undefined) => void;
    children?: ReactNode;
  };
