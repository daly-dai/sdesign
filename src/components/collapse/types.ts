import { ButtonProps } from 'antd/es/button/button';

export interface SCollapseProps extends ButtonProps {
  collapse?: boolean;
  setCollapse?: (collapse: boolean) => void;
  onExpand?: (collapse: boolean) => void;
}
