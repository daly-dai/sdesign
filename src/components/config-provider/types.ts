import { ReactNode } from 'react';

export type SConfigProviderType = {
  // 全部的字典
  globalDict?: Record<string, any>;
  // 上传组件接口地址地址
  uploadUrl?: string;
  children?: ReactNode;
  prefixCls?: string;
};

export interface ConfigContextProps {
  globalDict?: Record<string, Record<string, any> | any>;
  uploadUrl?: string;
  getPrefixCls: (suffixCls?: string, customizePrefixCls?: string) => string;
}
