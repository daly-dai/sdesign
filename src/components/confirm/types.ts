import { ButtonProps, PopconfirmProps } from 'antd';

import { tuple } from '@dalydb/sdesign/utils/types';

const ConfirmTypes = tuple('pop', 'modal');
export type ConfirmType = (typeof ConfirmTypes)[number];

export interface SConfirmProps {
  onConfirm?: () => void; // 确认的回调函数
  onCancel?: () => void; // 取消的回调函数
  type?: ConfirmType;
  text?: string;
  title?: React.ReactNode;
  description?: React.ReactNode;
  children?: React.ReactNode;
  popConfirmProps?: PopconfirmProps; // 弹出框的属性传参
  buttonProps?: ButtonProps; // 按钮的属性传参
  disabled?: boolean;
}
