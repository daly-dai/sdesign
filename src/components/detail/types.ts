import { DescriptionsProps } from 'antd';
import { DescriptionsItemType } from 'antd/es/descriptions';
import React, { ReactNode } from 'react';

import { FileListProps, PageTitleProps } from '@daly/sdesign';
import { tuple } from '@daly/sdesign/utils/types';

const ItemTypes = tuple(
  'text',
  'empty',
  'file',
  'dict',
  'img',
  'rangeTime',
  'checkbox',
  'placeholder',
);

export type ItemType = (typeof ItemTypes)[number];

export interface DictReflect {
  label?: string;
  name?: string;
}

export type SDetailItemType = DetailItemType & {
  label?: ReactNode;
  name?: string | string[];
  render?: (value?: any, dataSource?: any) => ReactNode;
  span?: number;
  /** * @desc 隐藏该属性 */
  hidden?: boolean;
};

export type DetailItemType = {
  type?: ItemType;
  fileProps?: Partial<FileListProps>;
  dictReflect?: DictReflect;
  dictKey?: string;
  value?: any;
  dictMap?: Record<string, string> | any[] | null;
  render?: (value?: any, dataSource?: any) => ReactNode;
  dataSource?: Record<string, any>;
};

export type SDetailItem = SDetailItemType &
  Omit<DescriptionsItemType, 'children' | 'key'>;

export interface SDetailProps
  extends Omit<DescriptionsProps, 'items' | 'title'> {
  titleDesc?: ReactNode;
  titleAction?: ReactNode;
  dataSource?: Record<string, any>;
  items?: SDetailItem[];
  labelStyle?: React.CSSProperties;
  contentStyle?: React.CSSProperties;
  hasCardBg?: boolean;
  container?: React.ComponentType<any>;
  title?: string | ReactNode;
  /** @description dataSource中的key */
  detailName?: string;
}

export interface SDetailGroupItem {
  groupTitle?: string | ReactNode;
  groupTitleProps?: Omit<PageTitleProps, 'title'>;
  groupContainer?: React.ComponentType<any>;
  groupItems?: SDetailProps[];
  items?: SDetailItem[];
  itemProps?: Omit<SDetailProps, 'items' | 'dataSource'>;
  dataSource?: Record<string, any>;
  /*** @desc 隐藏该属性*/
  hidden?: boolean;
}

export interface SDetailGroupProps {
  items?: SDetailGroupItem[];
  dataSource?: Record<string, any>;
}
