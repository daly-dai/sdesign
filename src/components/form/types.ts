import {
  Cascader,
  Checkbox,
  ColProps,
  DatePicker,
  FormItemProps,
  FormProps,
  Input,
  InputNumber,
  Radio,
  RowProps,
  Slider,
  Switch,
  Table,
  TimePicker,
  TreeSelect,
  Upload,
} from 'antd';
import { NamePath } from 'antd/es/form/interface';
import React, { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import SCascader from '../cascader';
import SCheckGroup from '../check-group';
import SDatePicker from '../date-picker';
import SDatePickerRange from '../date-picker-range';
import SDependency from '../dependency';
import { RenderChildren } from '../dependency/types';
import SRadioGroup from '../radio-group';
import SSelect from '../select';
import SUpload from '../upload';
import InternalUpload from '../upload/instance';

import { RegKeyType } from '@dalydb/sdesign/types/reg';

export type FormFieldMapType = {
  input: typeof Input;
  inputNumber: typeof InputNumber;
  password: typeof Input.Password;
  textarea: typeof Input.TextArea;
  select: typeof SSelect;
  slider: typeof Slider;
  radio: typeof Radio;
  radioGroup: typeof SRadioGroup;
  switch: typeof Switch;
  treeSelect: typeof TreeSelect;
  upload: typeof Upload;
  SUpload: typeof InternalUpload;
  SUploadDragger: typeof SUpload.Dragger;
  SUploadPicture: typeof SUpload.Picture;
  datePicker: typeof DatePicker;
  SDatePicker: typeof SDatePicker;
  datePickerRange: typeof DatePicker.RangePicker;
  SDatePickerRange: typeof SDatePickerRange;
  timePicker: typeof TimePicker;
  timePickerRange: typeof TimePicker.RangePicker;
  checkbox: typeof Checkbox;
  checkGroup: typeof SCheckGroup;
  cascader: typeof Cascader;
  SCascader: typeof SCascader;
  table: typeof Table;
  dependency: typeof SDependency;
};

export type FormComType = keyof FormFieldMapType;

export type FormComPropsType = HTMLAttributes<object> &
  ComponentProps<FormFieldMapType[FormComType]>;

export type FormItemType = FormComType | 'placeholder';
export interface ItemsProps
  extends Omit<FormItemProps, 'label | name' | 'required'> {
  label?: ReactNode;
  name?: NamePath;
  style?: React.CSSProperties;
  // 组件的类型
  type?: FormItemType;
  // 依赖的字段，只在type为dependency时生效
  depNames?: string[];
  // 组件的配置项
  fieldProps?: FormComPropsType;
  // 自定义组件
  customCom?: ReactNode | RenderChildren<any>;
  // 校验规则
  regKey?: RegKeyType;
  // 自定义渲染
  render?: RenderChildren<any>;
  // 是否为必填
  required?: string | boolean;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
}

export interface SFormItems extends ItemsProps {
  colProps?: ColProps;
  hidden?: boolean;
}

export interface SFormProps extends FormProps {
  rowProps?: RowProps;
  children?: ReactNode;
  items?: SFormItems[];
  columns?: number;
  // 是否为必填,提示文字
  required?: string | boolean;
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
}

export type GroupItemsType = {
  container?: React.ComponentType<any>;
  columns?: number;
  title?: ReactNode;
  items?: SFormItems[];
  rowProps?: RowProps;
  // 嵌套的数据结构
  formName?: string;
};

// 分组表单
export interface SFormGroupProps extends FormProps {
  groupItems?: GroupItemsType[];
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  container?: React.ComponentType<any>;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
  /** @description  只读模式 */
  readonly?: boolean;
}

export interface SearchProps extends SFormProps {
  // 默认是否展开
  defaultExpand?: boolean;
  showExpand?: boolean;
  // 展开收起的回调函数
  onExpand?: (expand: boolean) => void;
  // 展开的行数
  expandLine?: number;
  actionNode?: ReactNode;
  /** @description 自定义组件容器 */
  container?: React.ComponentType<any>;
  isCard?: boolean;
}
