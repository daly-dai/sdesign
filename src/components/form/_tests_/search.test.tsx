import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import SForm from '../index';
import { SFormItems } from '../types';

describe('SForm.Search', () => {
  /**
   * TODO: 测试用例
   * - 测试props
   * 支持默认设置是否展示展开收起按钮showExpand
   * 支持设置默认是展开还是收起的状态 defaultExpand
   *  支持自定义容器 container
   *  支持动态计算列数
   *
   */

  const items: SFormItems[] = [
    {
      type: 'input',
      label: '查询输入框',
      name: 'key',
    },
    {
      type: 'select',
      label: '查询下拉框',
      name: 'key1',
    },
    {
      type: 'input',
      label: '查询输入框',
      name: 'key3',
    },
    {
      type: 'select',
      label: '查询下拉框',
      name: 'key4',
    },
    {
      type: 'select',
      label: '查询下拉框',
      name: 'key5',
    },
  ];

  it('should support config showExpand', () => {
    const { getByText } = render(<SForm.Search items={items} />);

    const expandBtn = getByText('展开');

    expect(expandBtn).toBeInTheDocument();
  });

  it('should support config showExpand false', () => {
    render(<SForm.Search showExpand={false} items={items} />);

    expect(screen.queryByText('展开')).not.toBeInTheDocument();
  });

  it('should support config defaultExpand', () => {
    render(<SForm.Search defaultExpand={false} items={items} />);
    expect(screen.queryByText('收起')).toBeInTheDocument();
  });

  it('should support config container', () => {
    const Container = ({ children }: any) => {
      return <div>222{children}</div>;
    };
    render(<SForm.Search items={items} container={Container} />);
    expect(screen.queryByText('222')).toBeInTheDocument();
  });

  it('should  support config dynamicColumns with showExpand true', () => {
    const { container } = render(<SForm.Search items={items} />);
    expect(container).toMatchSnapshot();

    const columnsItems = container.querySelectorAll('.ant-col-8');

    expect(columnsItems.length).toBe(3);
  });

  it('should  support config dynamicColumns with showExpand false', () => {
    const { container } = render(
      <SForm.Search showExpand={false} items={items} />,
    );

    const columnsItems = container.querySelectorAll('.ant-col-8');

    expect(columnsItems.length).toBe(6);
  });
});
