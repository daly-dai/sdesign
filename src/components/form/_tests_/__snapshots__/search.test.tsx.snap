// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SForm.Search > should  support config dynamicColumns with showExpand true 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
    style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    >
      <form
        class="ant-form ant-form-horizontal css-dev-only-do-not-override-3rel02"
      >
        <div
          class="ant-row ant-row-middle css-dev-only-do-not-override-3rel02"
          style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
        >
          <div
            class="ant-col ant-col-8 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
              style="margin-bottom: 0px;"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="key"
                    title="查询输入框"
                  >
                    查询输入框
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                      >
                        <input
                          class="ant-input css-dev-only-do-not-override-3rel02"
                          id="key"
                          placeholder="请输入"
                          type="text"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                fill-rule="evenodd"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-8 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-3rel02"
              style="margin-bottom: 0px;"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
                >
                  <label
                    class="ant-form-item-no-colon"
                    for="key1"
                    title="查询下拉框"
                  >
                    查询下拉框
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-outlined ant-select-in-form-item css-dev-only-do-not-override-3rel02 ant-select-single ant-select-allow-clear ant-select-show-arrow"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-controls="key1_list"
                              aria-expanded="false"
                              aria-haspopup="listbox"
                              aria-owns="key1_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="key1"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          >
                            请选择
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-8 css-dev-only-do-not-override-3rel02"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div>
              <div
                class="acss-1bvc4cc"
              >
                <div
                  class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                  style="column-gap: 12px;"
                >
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-link"
                      style="font-size: 14px; padding: 4px 0px;"
                      type="button"
                    >
                      <span>
                        展开
                      </span>
                      <span
                        aria-label="caret-down"
                        class="anticon anticon-caret-down"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="caret-down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="0 0 1024 1024"
                          width="1em"
                        >
                          <path
                            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
                      type="reset"
                    >
                      <span>
                        重 置
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        查 询
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
